File: solutions/java/MinimumSwipes2.java
public class MinimumSwipes2{
    static int minimumSwaps(int[] arr) {

        int min = 1;
        int swaps = 0;
        int supposedPlace;
        int currentPlace;
        int temp;
        
        for(int i = 0; i < arr.length; i++){
            supposedPlace = arr[i] - min;
            if(supposedPlace != i){
               while(supposedPlace != i){
                    currentPlace = supposedPlace;
                    supposedPlace = arr[supposedPlace] - min;
                    temp = arr[currentPlace];
                    arr[currentPlace] = arr[i];
                    arr[i] = temp;
                    swaps++;
               }
            }    
        }
        return swaps;
    }
}

File: solutions/kotlin/MinimumSwipes2.kt
fun minimumSwaps(arr: Array<Int>): Int {
    var min = 1
    var swaps = 0
    var supposedPlace = 0
    var currentPlace = 0
    
    for(i in arr.indices){
        supposedPlace = arr[i] - min
        if(supposedPlace == i) continue
        while(supposedPlace != i){
            currentPlace = supposedPlace
            supposedPlace = arr[supposedPlace] - min
            arr[i] = arr[currentPlace].also { arr[currentPlace] = arr[i] }
            swaps++
        }
    }
    
    return swaps
}